client/grpc-web-react-native-transport/src/index.ts

```diff
@@ -0,0 +1,139 @@
+import { grpc } from "@improbable-eng/grpc-web";
+
+export function ReactNativeTransport(init: grpc.XhrTransportInit): grpc.TransportFactory {
+  return (opts: grpc.TransportOptions) => {
+    return new ArrayBufferXHR(opts, init);
+  }
+}
+
+let awaitingExecution: Array<() => void> | null = null;
+
+function runCallbacks() {
+  if (awaitingExecution) {
+    const thisCallbackSet = awaitingExecution;
+    awaitingExecution = null;
+    for (let i = 0; i < thisCallbackSet.length; i++) {
+      try {
+        thisCallbackSet[i]();
+      } catch (e) {
+        if (awaitingExecution === null) {
+          awaitingExecution = [];
+          setTimeout(() => {
+            runCallbacks();
+          }, 0);
+        }
+        for (let k = thisCallbackSet.length - 1; k > i; k--) {
+          awaitingExecution.unshift(thisCallbackSet[k]);
+        }
+        throw e;
+      }
+    }
+  }
+}
+
+function debug(...args: any[]) {
+  if (console.debug) {
+    console.debug.apply(null, args);
+  } else {
+    console.log.apply(null, args);
+  }
+}
+
+function detach(cb: () => void) {
+  if (awaitingExecution !== null) {
+    awaitingExecution.push(cb);
+    return;
+  }
+  awaitingExecution = [cb];
+  setTimeout(() => {
+    runCallbacks();
+  }, 0);
+}
+
+class XHR implements grpc.Transport {
+  options: grpc.TransportOptions;
+  init: grpc.XhrTransportInit;
+  xhr: XMLHttpRequest;
+  metadata: grpc.Metadata;
+  index: 0;
+
+  constructor(transportOptions: grpc.TransportOptions, init: grpc.XhrTransportInit) {
+    this.options = transportOptions;
+    this.init = init;
+  }
+
+  onProgressEvent() { }
```

Comment:
This is the only thing I'm not happy with.

AFAIK this is the only change from the XHR class copied from [`transports/http/xhr.ts`](https://github.com/improbable-eng/grpc-web/blob/master/client/grpc-web/src/transports/http/xhr.ts) which means it could easily be lost if we were ever de-dupliate the implementations.

Could you instead please restore the original `onProgressEvent()` implementation here, mark the method as protected in this base XHR class, and then override it with an empty implementation in `ArrayBufferXHR` implementation.

Thanks.